packages/core/src/components/icon/icon.tsx

```diff
@@ -17,50 +17,54 @@ export interface IIconProps extends IIntentProps, IProps {
      * Name of the icon (with or without `"pt-icon-"` prefix).
      * If `undefined`, this component will render nothing.
      */
-    iconName: IconName | undefined;
+    iconName: LegacyIconName | undefined;
```

Comment:
what's the point of distinguishing these from `IconName`?

Reply:
this is a `[WIP]` hack to make it compile. 
but it also preserves compatibility with the long `pt-icon-time` names.

Reply:
still doesn't answer the question -- _why_ are we distinguishing short from long names in the type? can't we just accept all of them

Reply:
I needed a type for the `Record<IconName, string[]>` in `iconSvgPaths` that _didn't_ include the long names so I wouldn't have to duplicate each short name key.* So I [split the old `IconName` enum in two](https://github.com/palantir/blueprint/pull/2028/files#diff-85d2ec5657eb35bcda535552e4063ffeR114) with `LegacyIconName = IconName | ...pt-long-names`

\* - Records with union types are sweet because they enforce that all valid keys have assigned values! Basically ensures that every defined icon has an associated file.

Reply:
@adidahiya what should we do about this interface? are you okay with dropping support for the long names?